### PR TITLE
add unfold,fold,bicubic,qr,eye

### PR DESCRIPTION
**Pull request by Exusial**
fold and bicubic need improving.

### CODE REVIEW

**Comment by cjld** [comment]
Thanks for your contribution~ Bravo~
 Could you please add some tests like `python/jittor/test/test_linalg.py`, and some changes seem to be automatically generated by editor(such as add space and enter), could you please remove that?

File: python/jittor/misc.py [hunk 1]
```diff
@@ -993,6 +993,10 @@ def grad(self, grad_a):
         dim+=len(a.shape)
     return func(a, dim)
 
+def eye(n):
+    x = jt.ones((n))
+    return x.reindex([n,n],["i0"],overflow_conditions=[f'i0!=i1'])
+
```
Comment (Gword) [inline python/jittor/misc.py:999]: eye似乎已经有了 在init.py

File: python/jittor/test/test_bicubic.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# ***************************************************************
+# Copyright (c) 2021 Jittor. All Rights Reserved.
+# Maintainers:
+#     Guoye Yang <498731903@qq.com>
+#     Dun Liang <randonlang@gmail.com>.
```
Comment (Gword) [inline python/jittor/test/test_bicubic.py:5]: 改一下这个matainer？

File: python/jittor/test/test_fold.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# ***************************************************************
+# Copyright (c) 2021 Jittor. All Rights Reserved.
+# Maintainers:
+#     Guoye Yang <498731903@qq.com>
+#     Dun Liang <randonlang@gmail.com>.
```
Comment (Gword) [inline python/jittor/test/test_fold.py:5]: 改一下

File: python/jittor/test/test_linalg.py [hunk 4]
```diff
@@ -8,49 +8,52 @@
...
     import autograd.numpy as anp
     from autograd import jacobian
+
     has_autograd = True
 except:
     has_autograd = False
 
+
 @unittest.skipIf(not has_autograd, "No autograd found.")
 class TestCodeOp(unittest.TestCase):
```
Comment (Gword) [inline python/jittor/test/test_linalg.py:27]: TestLinalgOp？

File: python/jittor/nn.py [hunk 5]
```diff
@@ -1282,49 +1393,103 @@ def __init__(self, *args):
...
+    if not isinstance(stride,tuple):
+        stride = (stride,stride)
+    n,cl,num = X.shape
+    area = kernel_size[0] * kernel_size[1]
+    block_nums = []
+    for i in range(2,4):
+        block_nums.append((output_size[i-2]+2*padding[i-2]-dilation[i-2]*(kernel_size[i-2]-1)-1) // stride[i-2]+1)
+    output = X.reindex_reduce("add",[n,cl // area,output_size[0]+2*padding[0],output_size[1]+2*padding[1]],["i0",f"i1/{area}",f"i2/{block_nums[1]}*{stride[0]}+(i1%{area})/{kernel_size[1]}*{dilation[0]}",f"i2%{block_nums[1]}*{stride[1]}+(i1%{area})%{kernel_size[1]}*{dilation[1]}"])
+    return output[:,:,padding[0]:padding[0]+output_size[0],padding[1]:padding[1]+output_size[1]]
+
```
Comment (Gword) [inline python/jittor/nn.py:1494]: 是不是只加了fold unfold，只留这个两个函数的修改就好了，别的撤销掉吧。可以从master的nn.py拷过来再贴上fold unfold。

**Pull request merged**